examples/grpc-bidirectional-streaming-copy/grpc_bidirectional_streaming.md

```diff
@@ -0,0 +1,54 @@
+# Bidirectional streaming RPC copy
+
+The gRPC server connector exposes the gRPC service over HTTP2. In a gRPC bidirectional streaming scenario, the gRPC service and the client operate when each other sends a sequence of messages using a read-write stream. In such scenarios, the two streams operate independently. Therefore, clients and servers can read and write in any order.
+
+>**Info:** For more information on the underlying module, see the [`grpc` module](https://lib.ballerina.io/ballerina/grpc/latest/).
+
+## Generate the service definition
+
+1. Create a new directory (e.g., `grpc_bidirectional_streaming`).
+
+2. Create a new Protocol Buffers definition file `grpc_bidirectional_streaming.proto` inside the `grpc_bidirectional_streaming` directory, and add the service definition below to it.
+
+    ::: code grpc_bidirectional_streaming.proto :::
+
+3. Navigate to the `grpc_bidirectional_streaming` directory, and run the command below for stub generation.
+
+    ::: code grpc_bidirectional_streaming.out :::
+
+Once you run the command, the `grpc_bidirectional_streaming_pb.bal` file gets generated inside the `stubs` directory. 
+
+>**Info:** For more information on how to use the Ballerina Protocol Buffers tool, see the <a href="https://ballerina.io/learn/by-example/proto-to-ballerina.html">Proto to Ballerina</a> example.
+
+## Implement and run the service
+
+1. Create a Ballerina package (e.g., `grpc_chat_service`) by executing the `bal new grpc_chat_service` command.
+
+>**Tip:** Delete the `main.bal` file created by default as it is not required for this example.
+
+2. Copy the generated `grpc_bidirectional_streaming_pb.bal` file from the `stubs` directory to the  `grpc_chat_service` package.
+
+3. Create a new `grpc_bidirectional_streaming_service.bal` Ballerina file inside the `grpc_chat_service` package and add the service implementation below.
```

Comment:
@daneshk Why should someone delete the generated main.bal file and create another file to have this main method? Isn't it confusing and not needed?

Reply:
I got an error when I had the `main.bal` file undelete together with the BAL file of the example. @dilanSachi

Reply:
@praneesha Users can copy the BBE content directly to their project's bal file. In the offline call, you already had 2 files which contain 2 main methods. That's why one file was needed to be deleted.

Reply:
@dilanSachi @daneshk - So, which is the most recommended method, to delete the `main.bal` file, or rename and update it?

Reply:
No need rename right? We can ask the user to just copy the content

Reply:
Since this is a service, maybe we can ask the user to use `bal new grpc_bidirectional_streaming_service -t service` and then update the service file content

Reply:
We have added instructions to have the BAL file name as `Create a new `grpc_bidirectional_streaming_service.bal` file inside the `grpc_chat_service` package and add the service implementation below.` as a best practice. In that case, we will have to ask the users to rename it.

Reply:
We asked to delete the main.bal to avoid confusion and be consistent in both service and client impl. In both scenarios, we ask to copy the content to a bal file and run. it makes sense to update the generated main.bal file on the client-side, but service side, we have to remove all content and replace it with example code. Shall we keep it as it is?

Reply:
Resolving as per Danesh's comment above.